File: CSC6301/Module2/Project 2/BubbleSort.java
/**
 * Simple class that provides an implementation of the BubbleSort sorting algorithm
 * for sorting an array of integers in an ascending order.
 * @author James Zafiri
 * @version 1.0.0
 * @since Week 2 of CSC6031
 */
public class BubbleSort {
    /**
     * Sorts a given array of integers using the bubble sort algorithm.
     * @param arr the array of integers that will be sorted
     */
    public static void bubbleSort(int[] arr) {
        int n = arr.length;
        boolean swapped;

        // initial loop through array
        for (int i = 0; i < n - 1; i++) {
            swapped = false;

            // second loop to compare integers in array
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    // swapping positions above
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swapped = true;
                }
            }

            // if there were no swaps made, then the array is already sorted
            if (!swapped) {
                break;
            }
        }
    }
    /**
     * This method will print out the elements of an array
     * @param args the array given
     */
    public static void printArray(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    /**
     * Main method of the class to test the BubbleSort algorithm
     * @param args the array that is given
     */
    public static void main(String[] args) {
        int[] myArray = {2, 45, 37, 21, 31, 50, 29, 22, 67, 88, 56};

        System.out.println("Array before sorting:");
        printArray(myArray);

        bubbleSort(myArray);

        System.out.println("Array after sorting:");
        printArray(myArray);
    }
}

File: CSC6301/Module7/bubblesort.py
# Bubble sort in Python
def bubbleSort(a):
    # loop to access each array element
    for i in range(len(a)):
        # loop to compare array elements
        for j in range(0, len(a)-i-1):
            # compare two adjacent elements
            if a[j] > a[j+1]:
                # swapping elements if elements not in order
                a[j+1], a[j] = a[j], a[j+1]
        for k in range(len(a)-1,len(a)-(i+2),-1):


            if (a[k] < a[k-1]):
                print("Error at i =", i)
                break
            
def main():
    a = [6,2,8,4,5,1,7,3]
    bubbleSort(a)
    print(a)
    
main()